schema/schema.go

```diff
@@ -320,6 +321,56 @@ func (ta *Table) fetchColumnsViaSqlDB(conn *sql.DB) error {
 	return r.Err()
 }
 
+// isMySQL8OrAbove checks if the given version string represents MySQL 8.0 or above.
+// It handles various MySQL version formats including:
+// - "8.0.25"
+// - "8.4.0"
+// - "8.0.25-log"
+// - "8.0.25-0ubuntu0.20.04.1"
+func isMySQL8OrAbove(version string) bool {
+
+	// Extract the version number before any additional info (like "-log", "-ubuntu", etc.)
+	versionParts := strings.Fields(version)
+	if len(versionParts) == 0 {
+		return false
+	}
+
+	// Take the first part and remove any suffix after hyphen
+	versionStr := strings.Split(versionParts[0], "-")[0]
+
+	// Use semver comparison for accurate version checking
+	if eq, err := mysql.CompareServerVersions(versionStr, "8.0.0"); err == nil && eq >= 0 {
+		return true
+	}
+
+	return false
+}
+
+func hasInvisibleIndexSupport(conn mysql.Executer) bool {
+	versionQuery := "SELECT VERSION()"
```

Comment:
Maybe we can use the version from the handshake and skip the extra roundtrip?

Reply:
Similar comment as above ^ Instead of comparing version, I think using show index to check for the Visible column is enough to see if it has invisible index support. I removed the logic for comparing versions. What do you think?

Reply:
I think this is much better. This would make it easier to support the same for MySQL, TiDB, MariaDB, etc. without too many version checks.